src/irr/video/COpenGLSpecializedShader.h

```diff
@@ -23,21 +25,48 @@ class COpenGLSpecializedShader : public IGPUSpecializedShader
         core::smart_refctd_dynamic_array<uint8_t> binary;
     };
 
-    COpenGLSpecializedShader(const asset::ICPUBuffer* _spirv, const asset::ISpecializationInfo* _specInfo, const asset::CIntrospectionData* _introspection);
+    COpenGLSpecializedShader(size_t _ctxCount, const asset::ICPUBuffer* _spirv, const asset::ISpecializationInfo* _specInfo, const asset::CIntrospectionData* _introspection);
 
-    //! @returns GL name or zero if already compiled once or there were compilation errors.
-    GLuint compile(uint32_t _GLSLversion);
+    inline GLuint getGLnameForCtx(const void* _ctx)
```

Comment:
make this a driver method, may use this for other stuff.

Reply:
as in get the ID

I see it working like this
```
COpenGLDriver* driver;
auto id = driver->getContextID(std::thread::id); /* or SAuxContext::contextID */
GLuint shaderHandle = specialized_shader->getGLnameForCtxID(id);
```

otherwise I see this becoming a problem when contexts are created and destroyed.

Btw main context will always be ID = 0  so first compilation (not from binary) can be stored there

Reply:
it's shader-local for a reason, i cant make it driver method because it searches shader's member array

Reply:
lets have a global ID between 0 and the max context count-1 that is assinged to context

Reply:
resolved